file/file_prefetch_buffer.h

```diff
@@ -31,6 +32,31 @@ namespace ROCKSDB_NAMESPACE {
 struct IOOptions;
 class RandomAccessFileReader;
 
+struct ReadaheadParams {
+  ReadaheadParams() {}
+
+  // The initial readahead size.
+  size_t initial_readahead_size = 0;
+
+  // The maximum readahead size.
+  // If max_readahead_size > readahead_size, then readahead size will be doubled
+  // on every IO until max_readahead_size is hit. Typically this is set as a
+  // multiple of initial_readahead_size. initial_readahead_size should be
+  // greater than equal to initial_readahead_size.
+  size_t max_readahead_size = 0;
+
+  // If true, Readahead is enabled implicitly by rocksdb
+  // after doing sequential scans for num_file_reads_for_auto_readahead.
+  bool implicit_auto_readahead = false;
+
+  uint64_t num_file_reads = 0;
```

Comment:
Can't remove the num_file_reads_ because it's initialized and passed down from BlockPrefetcher and based on it's value we decide if we want to prefetch or not in FilePrefetchBuffer. It's value right now is either 0 or >= 2. However I will add a todo note to remove it when refactoring BlockPrefetcher. In that case we don't need to pass it down but we do need that variable.

Reply:
Works for me